### ISSUE TITLE
Support for NFS share

### ISSUE THREAD

**Issue by jotterson** [hashicorp/terraform-provider-azurerm#8711]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It would be very useful to have azurerm support for NFS file shares, which is currently in preview.

### New or Affected Resource(s)

* azurerm_storage_share

### Potential Terraform Configuration

```hcl

resource "azurerm_storage_share" "example" {
  name                 = "sharename"
  storage_account_name = azurerm_storage_account.example.name
  enabled_protocol = "NFS"
  root_access = "RootSquash"
}
```

### References

 * https://docs.microsoft.com/en-us/azure/storage/files/storage-files-how-to-create-nfs-shares?tabs=azure-cli

**Comment by jbinko** [comment]
@WodansSon , @tombuildsstuff 

I wanted to move this further and implement this, but it seems to me that it has dependency (and I guess requires changes first) on tombuildsstuff/giovanni layer.
Correct me if I’m wrong.
See:
terraform-provider-azurerm\azurerm\internal\services\storage\shim\shares_data_plane.go
terraform-provider-azurerm\azurerm\internal\services\storage\storage_share_resource.go

If this is the true, what is the process for changes across dependencies?

**Comment by jimmfarley** [comment]
We need this!

**Comment by jonallenedf** [comment]
Same.

**Comment by tombuildsstuff** [comment]
@jbinko this has a dependency on the Resource Manager feature which is in turn blocked on a permissions issue in the Azure API (where the permissions for a "Contributor" Role Assignment created ~18m ago and created today have different permissions, but no visible difference) - as such we're blocked until that's fixed (but we're working with the team to get that fixed) - since this unfortunately caused issues last time we attempted switching.

**Comment by m4r74** [comment]
Are there any updates on this one? 
And also could the dependency on the RM feature be linked to this issue for easier tracking please?

**Comment by RKSelvi** [comment]
Any updates on this one? Created 'file storage' storage account for nfs file share but could not create via in TF

**Comment by markthebault** [comment]
it will be nice to have this

**Comment by vaclav-dvorak** [comment]
This is now possible through portal. Would be nice to have this in TF too.

**Comment by jrblanno** [comment]
please provide this function! :)

**Comment by jeremybusk** [comment]
For the love provide this functionality unless you want us not to use TF for storage. This seems really basic.

**Comment by Ravi-Kashyap** [comment]
Seems like the change in request #13938 was approved.  Does it mean Terraform can be used for NFS?

![image](https://user-images.githubusercontent.com/26795040/139912314-1804de77-1ad6-4473-a05d-9a2cbcaf8d11.png)
